zcash_client_sqlite/src/wallet/init.rs

```diff
@@ -289,11 +292,26 @@ fn init_wallet_db_internal<P: consensus::Parameters + 'static>(
 
     // Now that the migration succeeded, check whether the seed is relevant to the wallet.
     if let Some(seed) = seed {
-        if !wdb
+        // We ignore `SeedRelevance::NoDerivedAccounts`;
+        match wdb
             .is_seed_relevant_to_any_derived_accounts(&seed)
             .map_err(sqlite_client_error_to_wallet_migration_error)?
         {
-            return Err(WalletMigrationError::SeedNotRelevant.into());
+            SeedRelevance::Relevant => (),
+            // Every seed is relevant to a wallet with no accounts; this is most likely a
+            // new wallet database being initialized for the first time.
+            SeedRelevance::NoDerivedAccounts
```

Comment:
I'm considering adding a `NoAccounts` variant to simplify this case, as I think the mobile SDKs will just want to do the exact same logic as here.

Reply:
I think having the enum is good; usually if you have the option to use an enum instead of a boolean it's a good idea to do so, just for better type safety.